### ISSUE TITLE
More informative variable names in AOTAutograd

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#110700]
### 🐛 Describe the bug

Two main improvements:

1. It may be that the input GraphModule to AOTAutograd has meaningful names. make_fx should preserve those names
2. When performing backwards, we every gradient should be associated with a name from the original. We should use a name like `grad_original_name` to make clear the association between primal and tangent

### Versions

main

cc @EikanWang @jgong5 @wenzhe-nrv @chauhang @penguinwu @bdhirsh @bobrenjc93 @aorenste @SherlockNoMad @msaroufim @anijain2305 @zou3519

**Comment by ezyang** [comment]
still applies

**Comment by zou3519** [comment]
A repro:
```py
import torch

@torch.compile(fullgraph=True, backend="aot_eager")
def f(inp):
    foo = inp.sin()
    bar = foo @ foo
    return bar

inp = torch.randn(3, requires_grad=True)
f(inp)
```

Running this script with TORCH_LOGS=graph_code shows the following Dynamo graph. This graph preserves the original input variable names:
```

DEBUG:torch._dynamo.output_graph.__graph_code:TRACED GRAPH
 ===== __compiled_fn_1 =====
 /home/rzou/dev/dcp11/pt-dcp11/torch/fx/_lazy_graph_module.py class GraphModule(torch.nn.Module):
    def forward(self, L_inp_: "f32[3][1]cpu"):
        l_inp_ = L_inp_
        
         # File: /home/rzou/dev/dcp11/pt-dcp11/foo.py:6 in f, code: foo = inp.sin()
        foo: "f32[3][1]cpu" = l_inp_.sin();  l_inp_ = None
        
         # File: /home/rzou/dev/dcp11/pt-dcp11/foo.py:7 in f, code: bar = foo @ foo
        bar: "f32[][]cpu" = foo @ foo;  foo = None
        return (bar,)
```
with TORCH_LOGS=aot it gives the following forward+backward graphs:
```
===== Forward graph 0 =====
/home/rzou/dev/dcp11/pt-dcp11/torch/fx/_lazy_graph_module.py class GraphModule(torch.nn.Module):
   def forward(self, primals_1: "f32[3][1]cpu"):
        # File: /home/rzou/dev/dcp11/pt-dcp11/foo.py:6 in f, code: foo = inp.sin()
       sin: "f32[3][1]cpu" = torch.ops.aten.sin.default(primals_1)
       
        # File: /home/rzou/dev/dcp11/pt-dcp11/foo.py:7 in f, code: bar = foo @ foo
       dot: "f32[][]cpu" = torch.ops.aten.dot.default(sin, sin);  sin = None
       return (dot, primals_1)
       

NFO:torch._functorch._aot_autograd.jit_compile_runtime_wrappers.__aot_graphs:TRACED GRAPH
===== Backward graph 0 =====
<eval_with_key>.1 class GraphModule(torch.nn.Module):
   def forward(self, primals_1: "f32[3][1]cpu", tangents_1: "f32[][]cpu"):
        # File: /home/rzou/dev/dcp11/pt-dcp11/foo.py:6 in f, code: foo = inp.sin()
       sin: "f32[3][1]cpu" = torch.ops.aten.sin.default(primals_1)
       
        # File: /home/rzou/dev/dcp11/pt-dcp11/foo.py:7 in f, code: bar = foo @ foo
       mul: "f32[3][1]cpu" = torch.ops.aten.mul.Tensor(tangents_1, sin);  tangents_1 = sin = None
       
        # File: /home/rzou/dev/dcp11/pt-dcp11/foo.py:7 in f, code: bar = foo @ foo
       add: "f32[3][1]cpu" = torch.ops.aten.add.Tensor(mul, mul);  mul = None
       
        # File: /home/rzou/dev/dcp11/pt-dcp11/foo.py:6 in f, code: foo = inp.sin()
       cos: "f32[3][1]cpu" = torch.ops.aten.cos.default(primals_1);  primals_1 = None
       mul_2: "f32[3][1]cpu" = torch.ops.aten.mul.Tensor(add, cos);  add = cos = None
       return (mul_2,)     
```
the variable names could be more descriptive

**Comment by alex1xu** [comment]
I took on part 2 and have a draft up https://github.com/pytorch/pytorch/pull/187685. It names the backward graph's returned gradients grad_<primal> (via the existing GradAOTOutput descriptor) so each gradient is tied to the input it differentiates. The name achievable today is grad_primals_N but if part 1 lands later these will upgrade (to ex: grad_inp). I wanted to align before I regenerate goldens whether this looks okay and should land first on its own or do both together?
